strawberry/types/fields/resolver.py

```diff
@@ -344,8 +344,12 @@ def is_graphql_generic(self) -> bool:
 
     @cached_property
     def is_async(self) -> bool:
-        return iscoroutinefunction(self._unbound_wrapped_func) or isasyncgenfunction(
-            self._unbound_wrapped_func
+        from asgiref.sync import SyncToAsync
+
+        return (
+            isinstance(self._unbound_wrapped_func, SyncToAsync)
+            or iscoroutinefunction(self._unbound_wrapped_func)
+            or isasyncgenfunction(self._unbound_wrapped_func)
```

Comment:
How about using asyncio.iscoroutinefunction() instead of inspect.iscoroutinefunction()? It was said that asyncio.iscoroutinefunction() function does two tests; it uses inspect.iscoroutinefunction() first, and if that test fails tests if the function is a function with the @acyncio.coroutine decorator applied. (https://stackoverflow.com/questions/36076619/test-if-function-or-method-is-normal-or-asynchronous)

Reply:
@erikwrede it was there already, what I PR'd there was a performance improvement: https://github.com/django/asgiref/pull/390 =P

Reply:
@shmoon-kr sounds good!

Reply:
@bellini666 oops the diff fooled me haha